src/BaseListView.php

```diff
@@ -1,158 +1,194 @@
 <?php
-/**
- * @link http://www.yiiframework.com/
- *
- * @copyright Copyright (c) 2008 Yii Software LLC
- * @license http://www.yiiframework.com/license/
- */
 
 namespace Yiisoft\Yii\DataView;
 
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Data\Paginator\PaginatorInterface;
+use Yiisoft\Data\Reader\CountableDataInterface;
+use Yiisoft\Data\Reader\DataReaderInterface;
+use Yiisoft\Data\Reader\FilterableDataInterface;
+use Yiisoft\Data\Reader\OffsetableDataInterface;
+use Yiisoft\Data\Reader\SortableDataInterface;
 use Yiisoft\Factory\Exceptions\InvalidConfigException;
 use Yiisoft\Html\Html;
-use yii\helpers\Yii;
-use yii\widgets\LinkPager;
-use yii\widgets\Widget;
-use Yiisoft\Arrays\ArrayHelper;
 use Yiisoft\I18n\MessageFormatterInterface;
+use Yiisoft\View\View;
+use Yiisoft\Yii\DataView\Widget\LinkPager;
+use Yiisoft\Yii\DataView\Widget\LinkSorter;
 
 /**
  * BaseListView is a base class for widgets displaying data from data provider
  * such as ListView and GridView.
- *
  * It provides features like sorting, paging and also filtering the data.
- *
- * For more details and usage information on BaseListView, see the [guide article on data widgets](guide:output-data-widgets).
- *
- * @author Qiang Xue <qiang.xue@gmail.com>
- *
- * @since 2.0
+ * For more details and usage information on BaseListView, see the [guide article on data
+ * widgets](guide:output-data-widgets).
  */
-abstract class BaseListView extends Widget
+abstract class BaseListView
```

Comment:
Why isn't it a widget anymore?

Reply:
Because GridView/ListView/DetailView cannot implement `begin` and `end` methods

Reply:
As far as I know, it's not required to implement these.

Reply:
Class must be follow LSP, but it can't (`begin` and `end` are inherit, but can't work correctly)

Reply:
Yes but that's not the problem of BaseListView but a problem of View package.

Reply:
Now, what's the solution?